src/osd/SnapMapper.h

```diff
@@ -298,7 +298,16 @@ class SnapMapper : public Scrub::SnapMapReaderI {
     update_bits(mask_bits);
   }
 
-  std::set<std::string> prefixes;
+  std::set<std::string>           prefixes;
```

Comment:
This set only gets mutated in update_bits, right?  Can we drop prefix_iter_invalidate and just invoke reset_prefix_itr() where you currently set prefix_itr_invalidate?

Reply:
Adding new objects to SnapMapper requires restarting the search.
It seems illogical to add objects to a snap after you start trimming it, but unless there is a guaranteed rule to prevent this we should keep using prefix_itr_invalidate.

Maybe a better approach will be to use an optimistic search ignoring add_oid/update_oid and add a quick scan of the full prefixes once we removed all objects owned by a given snap?
If nothing else was added we will run a single loop costing us 8 extra searches in  worst case scenario?

What do you think

Reply:
I'm fine with resetting the search.  I am specifically suggesting that the bool doesn't seem necessary -- can we simply reset the iterator directly?